docs/src/dev/langref.md

```diff
@@ -1,4 +1,275 @@
+
+All introduced identifiers are namespaced, with the namespace indicated by a sigil (a prefix character like `!` or `%`). For identifiers that refer to objects that can be referenced in part, each reference unambiguously indicates the size of the partial reference so that it is not necessary to consult the declaration to find it out.
+
+[llvm-langref]: https://llvm.org/docs/LangRef.html
+
+
+## Lexical structure {#lexical}
+
+The text IR has a conservative lexical structure, where the first character of a non-whitespace sequence of characters determines its lexical category.
+
+The byte encoding of the text IR is always [UTF-8].
+
+[utf-8]: https://www.unicode.org/versions/Unicode16.0.0/core-spec/chapter-2/#G11165
+
+
+### Whitespace {#whitespace}
+
+The only characters recognized as whitespace are `U+0020 SPACE`, `U+0009 CHARACTER TABULATION`, and `U+000A LINE FEED`. All of them may be used for separating other tokens, and `U+000A LINE FEED` is used to indicate the end of a syntactic construct.
+
+The file must end with a `U+000A LINE FEED` character.
+
+<div class="warning">
+
+The [prjunnamed-netlist] parser currently handles whitespace in a questionable and inconsistent way. This will be fixed in the future.
+
+</div>
+
+
+### Comments {#comment}
+
+```
+;<text>
+```
+
+Only line comments are supported. Any sequence of characters starting with `;` and ending with `U+000A LINE FEED` is ignored.
+
+
+### Strings {#string}
+
+```
+"<bytes>"
+```
+
+For example:
+
+```
+""
+"meow"
+"this\0ais a anewline"
+"\22quoted\22"
+"\5cescaped"
+```
+
+Strings are delimited by `"` on both ends and denote byte sequences (not character sequences). Within a string, `` \ `` followed by two hexadecimal digits `0`..`9` or `a`..`f` (lowercase only) corresponds to that particular byte, and any other character corresponds to its unique [UTF-8] encoding. No normalization is performed.
+
+
+### Decimal numbers {#decimal}
+
+```
+#<digits>
+```
+
+For example:
+
+```
+#0
+#10
+#-1
+```
+
+Decimal numbers start with a `#` sigil, followed by an optional `-` sign and a non-empty sequence of `0`..`9` digits. No whitespace is allowed between any of the characters.
+
+In some contexts, only non-negative decimal numbers, or decimal numbers up to a certain magnitude, may be accepted.
+
+
+### Constants {#constant}
+
+```
+<trits>
+```
+
+For example:
+
+```
+0
+1
+X
+100
+1XX0
+XXXX
+```
+
+Constants are made up of `0`, `1`, and `X` (uppercase only) characters. No whitespace is allowed between any of the characters.
+
+Constants denote bit vectors with some positions being "don't care" or "undefined" (depending on the context), and are written in the usual order where the most significant position is on the left. That is, `1010` has the same numeric value as `#10`, its first bit is `0` and last bit is `1`.
```

Comment:
This suggests that decimal numbers may be used in contexts where a constant is expected. Is this the case?

Reply:
No.